### PR TITLE
Add test cases for JSON resource marshaling

### PR DESCRIPTION
**Pull request by sagar23sj**
Resources Covered :
- GistFile

Helps with issue: #55

### CODE REVIEW

File: github/gists_test.go [hunk 1]
```diff
@@ -1017,3 +1017,27 @@ func TestGistsService_ListForks_withOptions(t *testing.T) {
 		return resp, err
 	})
 }
+
+func TestGistFile_marshall(t *testing.T) {
```
Comment (gmlewis) [inline github/gists_test.go:1021]:
```suggestion
func TestGistFile_Marshal(t *testing.T) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @sagar23sj !
One minor tweak, please.

File: github/gists_test.go (hunk 1 above)
Comment (sagar23sj) [reply github/gists_test.go]: Updated @gmlewis

**Review by gmlewis — APPROVED** [review]
Thank you, @sagar23sj !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**